File: src/main/java/ac/kr/deu/connect/luck/event/application/EventApplication.java
package ac.kr.deu.connect.luck.event.application;

import ac.kr.deu.connect.luck.common.entity.BaseEntity;
import ac.kr.deu.connect.luck.event.entity.Event;
import ac.kr.deu.connect.luck.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EventApplication extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Event event;

    private Long foodTruckId;

    @ManyToOne
    private User foodTruckManager;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    private String comment;

}

File: src/main/java/ac/kr/deu/connect/luck/event/application/entity/EventApplication.kt
package ac.kr.deu.connect.luck.event.application.entity

import ac.kr.deu.connect.luck.common.entity.BaseEntity
import jakarta.persistence.Embeddable
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Entity
class EventApplication(
    @EmbeddedId
    val id: ApplicationId,

    @Enumerated(EnumType.STRING)
    private val status: ApplicationStatus = ApplicationStatus.PENDING,

    private val comment: String? = null,
) : BaseEntity() {
    @Embeddable
    data class ApplicationId(
        val eventId: Long,
        val foodTruckId: Long,
    )
}
